### PR TITLE
feat: sqlite storage

### PR DESCRIPTION
**Pull request by sebastianquek**
TLDR: `pouchdb-adapter-asyncstorage` isn't well maintained, use `pouchdb-adapter-react-native-sqlite` instead.

## Issue
When doing this:
```
    const subscription = db?.documents
      .find()
      .sort("created")
      .$.subscribe(setDocuments);
```
and updating the `documents` collection (i.e. update a single document), the subscription only returns the document that was updated.

## Solution
Some people have faced similar issues: https://github.com/pouchdb/pouchdb/issues/7293 and looks to be a react native related problem.
A new package that relies on sqlite was mentioned: https://www.npmjs.com/package/pouchdb-adapter-react-native-sqlite. However, it relies on `react-native-sqlite-2` which doesn't work with Expo (https://github.com/craftzdog/react-native-sqlite-2/issues/22). Using  Expo's SQLite instead works: https://docs.expo.io/versions/latest/sdk/sqlite/

Everything looks to work fine, sorting works as expected, but may need further testing to be sure.

## Additional references
https://dev.to/craftzdog/hacking-pouchdb-to-use-on-react-native-1gjh

### CODE REVIEW

**Comment by john-dot-oa** [comment]
## Application
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=40

## Storybook
Expo QR
Published to https://exp.host/@dlt/identitywallet?release-channel=storybook-40
